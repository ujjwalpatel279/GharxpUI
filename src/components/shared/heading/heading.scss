// ========================================================================
// COMPONENTS/SHARED/HEADING
// ========================================================================

@import '~sassHelpers';

.c-heading {
  color: color(grey5);
  font-weight: 600;

  &--h1 {
    padding-bottom: spacer(4);
    font-size: font-size(5);
  }

  &--h2 {
    font-size: font-size(4);
  }

  &--h3 {
    font-size: font-size(3);
  }
}
