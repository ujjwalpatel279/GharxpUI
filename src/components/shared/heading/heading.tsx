import React, { FunctionComponent, ReactElement } from 'react';

interface HeadingProps {
  level: 'h1' | 'h2' | 'h3' | 'h4';
  title: string;
  mb?: 1 | 2 | 3 | 4 | 5 | 6;
  extendClass?: string;
}

import './heading.scss';

export const Heading: FunctionComponent<HeadingProps> = ({ level, title, mb, extendClass }): ReactElement => {
  const HeadingLevel = level;

  let classes = `c-heading c-heading--${level}`;

  if (mb) {
    classes += ` u-mb-${mb}`;
  }
  if (extendClass) {
    classes += ` ${extendClass}`;
  }

  return <HeadingLevel className={classes}>{title}</HeadingLevel>;
};
