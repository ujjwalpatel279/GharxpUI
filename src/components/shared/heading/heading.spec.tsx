import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Heading } from './heading';

describe('Container Component', () => {
  afterEach(cleanup);

  it('renders the level 1 heading', () => {
    const { container } = render(<Heading level="h1" title="Heading Level 1" />);

    const headingLevel = container.querySelector('h1');

    expect(container.firstChild).toMatchSnapshot();
    expect(headingLevel).toBeInTheDocument();
  });

  it('renders the level 2 heading', () => {
    const { container } = render(<Heading level="h2" title="Heading Level 2" />);

    const headingLevel = container.querySelector('h2');

    expect(container.firstChild).toMatchSnapshot();
    expect(headingLevel).toBeInTheDocument();
  });

  it('renders the level 3 heading', () => {
    const { container } = render(<Heading level="h3" title="Heading Level 3" />);

    const headingLevel = container.querySelector('h3');

    expect(container.firstChild).toMatchSnapshot();
    expect(headingLevel).toBeInTheDocument();
  });
});
