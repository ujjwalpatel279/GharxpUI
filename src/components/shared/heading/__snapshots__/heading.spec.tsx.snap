// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders the level 1 heading 1`] = `
<h1
  class="c-heading c-heading--h1"
>
  Heading Level 1
</h1>
`;

exports[`Container Component renders the level 2 heading 1`] = `
<h2
  class="c-heading c-heading--h2"
>
  Heading Level 2
</h2>
`;

exports[`Container Component renders the level 3 heading 1`] = `
<h3
  class="c-heading c-heading--h3"
>
  Heading Level 3
</h3>
`;
